src/gui_common/AddWindowReorderingSupportToSiblings.cs

```diff
@@ -0,0 +1,328 @@
+    public void OnWindowReorder(CustomDialog window)
+    {
+        // Get a sibling that is an ancestor of this window
+        Node targetSibling = connectedWindows[window];
+
+        UpdateTopSibling();
+
+        int topSiblingIndex = topSibling!.GetIndex();
+
+        if (topSiblingIndex == targetSibling.GetIndex())
+        {
+            // This window is already on the top
+            return;
+        }
+
+        // Put the sibling on the top
+        parent.MoveChild(targetSibling, topSiblingIndex);
+
+        topSibling = targetSibling;
+
+        // For unexplained reasons this has to be here to update the order visually
+        window.SetAsToplevel(false);
+        window.SetAsToplevel(true);
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        if (disposing)
+        {
+            foreach (var path in WindowReorderingPaths)
+                path.Dispose();
+        }
+
+        base.Dispose(disposing);
+    }
+
+    /// <summary>
+    ///   Used to find window reordering nodes automatically.
+    /// </summary>
+    /// <returns>
+    ///   Pairs containing window reordering nodes and what sibling of theirs is an ancestor of this class.
+    /// </returns>
+    private static IEnumerable<(AddWindowReorderingSupportToSiblings Node, Node Sibling)> GetWindowReorderingAncestors(
+        Node startingNode, int maxSearchDepth)
+    {
+        Node childOfAncestor = startingNode;
+        Node ancestor = childOfAncestor.GetParent();
+
+        for (int i = 0; i < maxSearchDepth; ++i)
+        {
+            if (ancestor is null)
+                break;
+
+            // Try to get the window reordering node
+            var windowReorderingSupportNode = ancestor.GetNodeOrNull
+                <AddWindowReorderingSupportToSiblings>(nameof(AddWindowReorderingSupportToSiblings));
+
+            if (windowReorderingSupportNode != null)
+            {
+                // The window reordering node exists, return it
+                yield return (windowReorderingSupportNode, childOfAncestor);
+
+                // Stop searching
+                break;
+            }
+
+            childOfAncestor = ancestor;
+            ancestor = childOfAncestor.GetParent();
+        }
+    }
+
+    /// <summary>
+    ///   Used to find ancestors that are siblings of window reordering nodes set by manual paths.
+    /// </summary>
+    /// <returns>
+    ///   Pairs containing window reordering nodes and what sibling of theirs is an ancestor of this class.
+    /// </returns>
+    private static IEnumerable<(AddWindowReorderingSupportToSiblings Node, Node Sibling)> GetWindowReorderingAncestors(
+        Node startingNode, Array<NodePath> windowReorderingNodePaths)
+    {
+        Node childOfAncestor = startingNode;
+        Node ancestor = childOfAncestor.GetParent();
+
+        System.Collections.Generic.Dictionary
+            <Node, AddWindowReorderingSupportToSiblings> windowReorderingNodesWithParents = new();
```

Comment:
If you just need to know whether a node is underneath another path, you can get the path of the current node and then (after converting to strings) just check if that node's path starts with any of the ancestor paths to start. I've used this approach in the focus grabber system to avoid having to loop a bunch of parent nodes instead of doing just one string operation.

Reply:
I wrote a new logic for that and I added checks if the path is correct. There was also a bug because the paths should be relative to the startingNode so I rewrote the comment about this propertly and added a check to automatic search to skip the first step if the startingNode is reordering node to prevent returning itself. (In case of reordering node I passed its parent and I changed it to pass itself)

Reply:
In case it helps I wrote a helper method for converting paths to absolute paths. So you could use that here as well. I think it's also a good idea to add caching when converting the paths to strings to avoid unnecessary operations. Though the cached version will need to be cleared when the node exits the tree as it might get reattached at a different place meaning that relative paths will have a different absolute results after that.

Reply:
What helper method? If this is to skip getting the node and just getting the path then it's not useful in this case because the nodes are also needed. GetPath() is enough in this case, I think.

This code is used only after entering the tree. If you want the code to be more performant then the paths can point to the windows ancestor instead of the window reordering node. This way we can skip the manipulations with absolute paths and can just look for "..\AddWindowReorderingSupportToSiblings" from there.

I have already implemented it so you can take a look at it.

Also the only reason to convert paths to strings here is to check if the path is set correctly.
I think that I can check the nodes themselves instead of using Contains on strings so the only string would be the starting node path and each starting node is needed only once after entering the tree so there is no reason to cache it.

Reply:
Before I take a look, you could look at how FocusGrabber class does it, especially how skipOverridingStringConverted is generated and regenerated. Having the paths precomputed is what allows skipping parent node processing as you can just use a simple string starts with to compare if something is an ancestor of some node instead of needing to go from C# -> C++ -> C# -> C++ which is the case when needing to call a bunch of Godot methods in a row.

Reply:
I was wrong about what I wanted to do. StartsWith definitely makes more sense than using Contains.
I need to convert only a starting node to string and this is done before the loop, otherwise I pass a NodePath directly to StartsWith, not sure what this does under the hood.
Outside of path validity checks there is no need to convert the paths to strings, all I need is GetName on the starting node's path.

I think that the code from FocusGrabber is not needed in this case, but I am not sure if I understand it. I think that the code is good enough how it is so I will let it on you to suggest some specific changes.